src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParser.cs

```diff
@@ -183,6 +184,16 @@ internal SqlInternalConnectionTds Connection
             }
         }
 
+        private static bool EnableTruncateSwitch
+        {
+            get
+            {
+                bool value;
+                value = AppContext.TryGetSwitch(enableTruncateSwitch, out value) ? value : false;
```

Comment:
There is precedence for using context switches in SqlClient in the netfx code, which includes caching and defining default values. Take a look at LocalAppContextSwitches.cs on the netfx side.